From 09586e515d336fcaed96d410b954c62a73c6ce8b Mon Sep 17 00:00:00 2001
From: Myron Stowe <myron.stowe@redhat.com>
Date: Sun, 10 Aug 2014 18:20:52 -0400
Subject: [iommu] Change iommu driver to call unmap trace event

Message-id: <20140810182052.24055.29074.stgit@gir.stowe>
Patchwork-id: 87585
O-Subject: [RHEL7.1 PATCH 14/19] iommu: Change iommu driver to call unmap trace event
Bugzilla: 1127467
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Dutile <ddutile@redhat.com>
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1127467
Upstream Status: 3a50639ca4684476a69314811d89622d78c09448

commit 3a50639ca4684476a69314811d89622d78c09448
Author: Shuah Khan <shuah.kh@samsung.com>
Date:   Thu Aug 15 11:59:29 2013 -0600

    iommu: Change iommu driver to call unmap trace event

    Change iommu driver to call unmap trace event. This iommu_map_unmap class
    event can be enabled to trigger when iommu unmap iommu ops is called. Trace
    information includes iova, physical address (map event only), and size.

    Testing:
    Added trace calls to iommu_prepare_identity_map() for testing some of the
    conditions that are hard to trigger. Here is the trace from the testing:

           swapper/0-1     [003] ....     1.854102: unmap: IOMMU: iova=0x0000000

    Signed-off-by: Shuah Khan <shuah.kh@samsung.com>
    Signed-off-by: Joerg Roedel <joro@8bytes.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/iommu/iommu.c b/drivers/iommu/iommu.c
index ea49fe8..d8c53c7 100644
--- a/drivers/iommu/iommu.c
+++ b/drivers/iommu/iommu.c
@@ -892,6 +892,7 @@ size_t iommu_unmap(struct iommu_domain *domain, unsigned long iova, size_t size)
   unmapped += unmapped_page;
  }
 
+ trace_unmap(iova, 0, size);
  return unmapped;
 }
 EXPORT_SYMBOL_GPL(iommu_unmap);
-- 
1.7.1